### DISCUSSION TITLE
In One minute, how many link we can generate?

### DISCUSSION THREAD

**Discussion by IamAkashJoshi** [YOURLS/YOURLS#3128] [Q&A]
Hello There,

We are using this project and every day we need to generate around 1,00,000 links.

Today I observed that our system generates 200 links/minute and it's very slow.

So can you suggest to me that how can I generate 2000+ links/minute?

_Domein infomation_
 - Code is published on the digitaloccen.

_this is code sample we are using_
 
` 
string keyword = RandomString(6, true);
string urllong = url.Replace(" ", "%20");
service.CreateShortUrl(UrlEncoder.Default.Encode(urllong), auth, out YourlsCreateShortUrlResult result, keyword);
return string.IsNullOrEmpty(result.ShortUrl) ? string.Empty : result.ShortUrl;`
 
 
 Please help me with this ASAP.
 
 Thanks

**Comment by ozh** [comment]
Generating a short URL costs a couple SQL queries and one external HTTP request if you don't provide a title for the link and expect YOURLS to fetch it for you from the page itself.

So, the answer to your question is : how many SQL queries and HTTP requests can your server do in one minute.

**Reply by IamAkashJoshi** [reply]
- First, we can't provide the Title.
- There is no sql queries and http request is there.
_If you see below is the code which written in C# service :_  
`string keyword = RandomString(6, true);
string urllong = url.Replace(" ", "%20");
service.CreateShortUrl(UrlEncoder.Default.Encode(urllong), auth, out YourlsCreateShortUrlResult result, keyword);
return string.IsNullOrEmpty(result.ShortUrl) ? string.Empty : result.ShortUrl;`

**Reply by IamAkashJoshi** [reply]
@ozh if we provide a title and not provide the keyword, does that increase the speed?

**Reply by ozh** [reply]
You don't understand. There ARE a couple SQL queries. When you create a short URL, it is written (saved) in the DB = SQL queries.

My answer remains valid : how many links you can create in one minute = how many SQL queries and HTTP requests your server can perform in one minute.

**Reply by ozh** [reply]
Providing title = saving an external http request to fetch the title. So yes, it saves time.

Providing keyword = depends on your use case. Could affect speed or not.

**Reply by IamAkashJoshi** [reply]
@ozh my domain is related the sending a message (SMS) so in one go we send around 1,00,000 messages and at that time we need to generate a 1,00,000 unique short URL.

As i review the code, if we send 10,000 messages without URL then it only takes around 10 minutes to send.


but if i use shorturl then, its took 60+ minutes to send a 10,000 message.

**Reply by IamAkashJoshi** [reply]
@ozh please respond.

**Reply by LeoColomb** [reply]
> please respond.

Is this a joke, @IamAkashJoshi?

The answer provided by @ozh above is as complete as possible. If you rely on external services/libraries, you should ask their respective support.

**Reply by IamAkashJoshi** [reply]
@LeoColomb really this is the answer?

**Reply by ozh** [reply]
Well yes. You're asking a question about YOURLS, we respond about YOURLS... Not sure what we can do more.

**Reply by dgw** [reply]
Of course the process will take longer if you add more steps to it. Of course YOURLS can only run as fast as the hardware it's on allows.

@LeoColomb See #3015. OP asked this before and was apparently unsatisfied with the answers the first time around.
